### ISSUE TITLE
Question: How to "Using your data"

### ISSUE THREAD

**Issue by asindl** [run-llama/create-llama#448]
I am not sure, if that is the right forum to ask questions, but I will still post it. 
I got the llama chatbot up and running perfectly fine; I also added my own local tool, that can interact with my own databases.
Now, I wanted to experiment with that "Using your data" feature. I followed the readme instructions and I guess, I got everything ready to go (pdf in /data folder, doc_store, index_store, vector_store got generated). 
But I don't know, how to control, that the chatbot shall use my own local tools to interact with my databases or answer questions around my pdf.
For sure, it tries to phrase queries to run against my database instead of answering from the pdf.... Am I missing something?

**Comment by leehuwuj** [comment]
Hi @asindl, I think you are having a problem on prioritizing the tool for specific task/question. I'd recommend you firstly to update the system prompt (it'll also coming on our next release).

So your system prompt can look like this:
```
You are a helpful assistant who helps users with their questions. 
You have access to a knowledge base; always use the query engine tool to retrieve the facts to answer the user's question. 
If the information is not enough to answer, use <your local database tool name> to query additional data.
```

Also, if the data in the PDF and the database are different, then you should tune the prompt for the context of the data. e.g:
```
Use these tools to answer the user's question:
- query_engine_tool: To retrieve data about <your pdf file context>
- <your local tool name>: To retrieve data about <your database data context>
```

Instead of tuning system prompt. You can also tuning the description of your tool for your specific use case.

**Comment by asindl** [comment]
Thanks for the reponse @leehuwuj. That all makes a lot of sense. I am trying out your latter approach as we speak and might have a follow-up question: the document parsing and indexing is a built-in functionality of the `create-llama` app. Is the retrieval of the document content a built-in tool as well and how would I use that? In your latter approach, you indicate `query_engine_tool`. Is that the tool name already? Or is there a standard document interaction tool name? Or do I have to setup my own local tool for that?
The other problem I ran into: Azure OpenAI lately only allows 1024 characters for the description of tools, but since I added the full database schema into the description, that is way too long now...

**Comment by asindl** [comment]
I figured it out: after reading through this [thread ](https://community.openai.com/t/tool-calling-api-upgrade-1024-char-limit-is-limiting/951951/5), I moved the lengthy description to the description of the property of the function. The description of the function itself is limited to 1024 characters, as it is supposed to be an overview of the function.
With this and the changes you indicated above, the tool now distinguishes which task to execute!
Thanks again @leehuwuj
